src/Nethermind/Nethermind.Synchronization/Blocks/BlockDownloader.cs

```diff
@@ -296,7 +207,7 @@ public virtual async Task<long> DownloadBlocks(PeerInfo? bestPeer, BlocksRequest
                         blocksSynced++;
                     }
 
-                    currentNumber += 1;
+                    _currentNumber += 1;
```

Comment:
Could this get out of sync with actual block numbers being requested?

Reply:
Line 155 reset it and only one of this `DownloadBlocks` is executed at any time.